server/app/api/controllers/users.js

```diff
@@ -71,5 +92,44 @@ module.exports = {
         });
       }
     });
+  },
+  update: function(req, res, next) {    
+    userModel.findOneAndUpdate({ userName: req.params.userName }, 
+      {
+          $set: req.body 
+      },
+      {upsert: true}, 
```

Comment:
why do we need upsert?

Reply:
This is part of the syntax for findOneAndUpdate where upsert : upsert: bool - creates the object if it doesn't exist. defaults to false.  I have removed Upsert now